### PR TITLE
Freeze string literals and mutable constants

### PR DESCRIPTION
**Pull request by jage**
I did this with rubocop:

    rubocop --only Style/FrozenStringLiteralComment,Style/MutableConstant --auto-correct

Opted to not add rubocop at this stage, wanted to limit the amount of
work here.

### CODE REVIEW

File: spec/lib/twingly/url_spec.rb [hunk 1]
```diff
@@ -134,7 +136,7 @@ def valid_urls
     end
 
     context "when given ASCII input" do
-      let(:ascii_url) { "http://www.twingly.com/öあ".force_encoding("ASCII-8BIT") }
+      let(:ascii_url) { "http://www.twingly.com/öあ".dup.force_encoding("ASCII-8BIT") }
```
Comment (jage) [inline spec/lib/twingly/url_spec.rb]: This broke without the `.dup`, I think this is ok but feel free to suggest something if you know other ways to create the ascii-string.

**Comment by walro** [comment]
Performance benchmarks? 😉

**Comment by jage** [comment]
> Performance benchmarks? 😉

I'm guessing it's hardly noticeable compared to the Addressable/Public-suffix stuff going on under the hood, but if it would be cool to have two more rake tasks, one to measure speed and another to meassure allocated objects. Currently I think we're only profiling method calls IIRC.

**Comment by jage** [comment]
I think this should be extracted to a frozen constant as well, if I understand this correctly there will be a new Regex object on each `normalize_blogspot`-call currently.

https://github.com/twingly/twingly-url/blob/401cbaa49004fff882d4306ab1a84ba787e79cda/lib/twingly/url.rb#L208

**Comment by walro** [comment]
> I'm guessing it's hardly noticeable compared to the Addressable/Public-suffix stuff going on under the hood, but if it would be cool to have two more rake tasks, one to measure speed and another to measure allocated objects. Currently I think we're only profiling method calls IIRC.

I wasn't that serious, but yeah good ideas! :)

**Pull request merged**